File: src/main/java/com/artillexstudios/axminions/integrations/implementation/stacker/StackerIntegration.java
package com.artillexstudios.axminions.integrations.implementation.stacker;

import com.artillexstudios.axminions.integrations.Integration;
import org.bukkit.entity.Entity;

import java.util.OptionalLong;

public final class StackerIntegration extends Integration<StackerIntegrable> {

    public StackerIntegration() {
        this.register(new DefaultStackerIntegrable());
    }

    public long getStackSize(Entity entity) {
        for (StackerIntegrable integration : this.integrations()) {
            OptionalLong stackSize = integration.stackSize(entity);
            if (stackSize.isPresent()) {
                return stackSize.getAsLong();
            }
        }

        return 1;
    }
}

File: api/src/main/kotlin/com/artillexstudios/axminions/api/integrations/types/StackerIntegration.kt
package com.artillexstudios.axminions.api.integrations.types

import com.artillexstudios.axminions.api.integrations.Integration
import org.bukkit.Location
import org.bukkit.entity.Item
import org.bukkit.entity.LivingEntity
import org.bukkit.inventory.ItemStack

interface StackerIntegration : Integration {

    fun getStackSize(entity: LivingEntity): Long

    fun getStackSize(item: Item): Long

    fun dropItemAt(itemStack: ItemStack, amount: Int, location: Location)
}